packages/flutter/lib/src/material/ink_well.dart

```diff
@@ -983,10 +992,15 @@ class _InkResponseState extends State<_InkResponseStateWidget>
     updateHighlight(_HighlightType.pressed, value: true);
   }
 
-  void _handleTap(BuildContext context) {
+  void _handleTapUp(BuildContext context, [TapUpDetails details]) {
     _currentSplash?.confirm();
     _currentSplash = null;
     updateHighlight(_HighlightType.pressed, value: false);
+    if (details != null) {
+      if (widget.onTapUp != null)
+        widget.onTapUp(details);
+      return;
+    }
```

Comment:
```suggestion
  void _handleTapUp(BuildContext context, TapUpDetails details) {
    _currentSplash?.confirm();
    _currentSplash = null;
    updateHighlight(_HighlightType.pressed, value: false);
    if (widget.onTapUp != null)
      widget.onTapUp(details);
```

Also `Feedback` in L1004 should be controlled by `onTap != null || onTapUp != null` instead.